charts/immich/templates/machine-learning/deployment.yaml

```diff
@@ -3,37 +3,42 @@ apiVersion: apps/v1
 metadata:
   name: {{ template "common.fullname" . }}-machine-learning
   labels:
-    app.kubernetes.io/name: {{ template "common.name" . }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/component: "immich-machine-learning"
+    app.kubernetes.io/component: "machine-learning"
     {{- include "common.labels" . | nindent 4 }}
 spec:
   revisionHistoryLimit: 0
   replicas: {{ .Values.server.replicas }}
-  strategy:
-    type: Recreate
   selector:
     matchLabels:
-      app.kubernetes.io/name: {{ template "common.name" . }}
-      app.kubernetes.io/instance: {{ .Release.Name }}
-      app.kubernetes.io/component: "immich-machine-learning"
+      app.kubernetes.io/component: "machine-learning"
+      {{- include "common.selectorLabels" . | nindent 6 }}
   template:
     metadata:
       labels:
-        app.kubernetes.io/name: {{ template "common.name" . }}
-        app.kubernetes.io/instance: {{ .Release.Name }}
-        app.kubernetes.io/component: "immich-machine-learning"
-        {{- include "common.labels" . | nindent 8 }}
+        app.kubernetes.io/component: "machine-learning"
+        {{- include "common.selectorLabels" . | nindent 8 }}
     spec:
       containers:
         - name: server
           image: "{{ .Values.machineLearning.image.registry }}/{{ .Values.machineLearning.image.repository }}:{{ .Values.machineLearning.image.tag }}"
           imagePullPolicy: {{ .Values.machineLearning.image.pullPolicy }}
           env:
-            - name: IMMICH_PORT
-              value: "3003"
-            - name: TRANSFORMERS_CACHE
-              value: /cache
```

Comment:
Env variable does not exist according to docs: https://immich.app/docs/install/environment-variables#machine-learning

Could replace with MACHINE_LEARNING_CACHE_FOLDER but /cache is the default anyway